guild-factory/src/config.rs

```diff
@@ -23,6 +23,25 @@ pub trait ConfigModule {
         config_mapper.set(config_address);
     }
 
+    #[only_owner]
+    #[endpoint(upgradeConfigSc)]
+    fn upgrade_config_sc(&self, source_contract_address: ManagedAddress) {
+        let config_mapper = self.config_sc_address();
+        require!(!config_mapper.is_empty(), "Config SC not deployed");
+
+        let config_address = config_mapper.get();
+        let gas_left = self.blockchain().get_gas_left();
```

Comment:
I think `gas_left` should be computed after `self.get_default_code_metadata` to be accurate.

Reply:
Nah, it doesn't matter. It only matters in cases where you actually use the whole gas on upgrade, and you get over the limit. Even if I were to use `u64::MAX` as gas limit, it would still work. (at least it used to)